### PR TITLE
fix(ui,api): dryrun endpoint send result through the websocket

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Review by fsamin — APPROVED** [review]

**Review by yesnault — APPROVED** [review]

File: sdk/event_workflow.go [hunk 1]
```diff
@@ -34,3 +34,11 @@ type EventWorkflowPermissionDelete struct {
 	WorkflowID int64           `json:"workflow_id"`
 	Permission GroupPermission `json:"group_permission"`
 }
+
+// EventRetentionWorkflowDryRun represents the vent when execution dry run on workflow retention
+type EventRetentionWorkflowDryRun struct {
+	Runs          []WorkflowRunToKeep `json:"runs"`
+	Status        string              `json:"status"`
+	Error         string              `json:"error"`
+	RunsAnalyszed int64               `json:"nb_runs_analyzed"`
```
Comment (richardlt) [inline sdk/event_workflow.go:43]:
```suggestion
	RunsAnalyzed int64               `json:"nb_runs_analyzed"`
```

File: ui/src/app/event.service.ts [hunk 2]
```diff
@@ -67,6 +67,19 @@ export class EventService {
         }
     }
 
+    subscribeToWorkflowPurgeDryRun(projectKey: string, workflowName: string, username: string) {
+        this.updateFilters(this.currentFilters.concat(<WebsocketFilter>{
+            type: WebsocketFilterType.WORKFLOW_RETENTION_DRYRUN,
+            project_key: projectKey,
+            workflow_name: workflowName,
+            username: username
```
Comment (richardlt) [inline ui/src/app/event.service.ts]: Remove username

File: engine/api/websocket.go [hunk 3]
```diff
@@ -335,9 +344,9 @@ func (c *websocketClient) updateEventFilters(ctx context.Context, db gorp.SqlExe
 
 	// Update client filters
 	c.mutex.Lock()
-	c.filters = make(webSocketFilters)
+	c.filters = make(webSocketFilters, 0, len(fs))
 	for i := range fs {
```
Comment (richardlt) [inline engine/api/websocket.go:348]: Maybe a copy should be better than this loop

File: engine/api/websocket.go [hunk 4]
```diff
@@ -302,6 +299,18 @@ func (c *websocketClient) updateEventFilters(ctx context.Context, db gorp.SqlExe
 			return err
 		}
 		switch f.Type {
+		case sdk.WebsocketFilterTypeDryRunRetentionWorkflow:
```
Comment (richardlt) [inline engine/api/websocket.go:302]: Can be grouped with case sdk.WebsocketFilterTypeWorkflow, sdk.WebsocketFilterTypeAscodeEvent:

**Review by richardlt — APPROVED** [review]

**Pull request merged**